force-app/main/default/classes/Chainable.cls

```diff
@@ -98,13 +133,13 @@ public abstract class Chainable {
 		return this;
     }
 
+	@TestVisible
+	private List<Chainable__e> unlink() {
 
-	public List<ChainableDeferredEvent__e> unlink() {
-
-		List<ChainableDeferredEvent__e> deferEvents = new List<ChainableDeferredEvent__e>();
+		List<Chainable__e> deferEvents = new List<Chainable__e>();
```

Comment:
I am a bit confused by you switching between "deferred" and "defer". Please check if all uses of that name actually make sense in its form: meaning is this a defer event (an event to defer something) or is it a deferred event.

Reply:
In the case of the events, with the previous event naming, when the event was published it's already conceptually deferred since the original transaction has already over, however the variable it's still called `deferEvents` since at the time of event creation those are still pending to be deferred, and the concept they want to defer are the execution of the chain links.

There are other places when those 2 names also differ conceptually, which are essentially in the `getDeferArgs` and `getDeferShared` vs `setDeferredArgs` and `setDeferredShared`, the first two are used to prepare the instance for deferring and the latter two are used to rebuild the instance appropriate in the deferred context.